packages/pwa-kit-react-sdk/src/ssr/server/react-rendering.js

```diff
@@ -252,6 +264,7 @@ const renderApp = (args) => {
 
     // Remove the stacktrace when executing remotely as to not leak any important
     // information to users about our system.
+    const error = renderError || appStateError
```

Comment:
I'm almost inclined to say that we should reverse this or statement so that the appStateError take precedence. The way I came to this conclusion was to ask myself how I would word an inline comment on why I'm making the decision that I'm making. This is what I felt came more natural:

```
// Return the first error encountered during the rending pipeline.
const error = appStateError || renderError
```

Let me know if you have a better reason for it to be the current order.